packages/jest-resolve-dependencies/src/__tests__/dependency_resolver.test.js

```diff
@@ -39,66 +39,135 @@ beforeEach(() => {
-  const resolved = dependencyResolver.resolve(
-    path.resolve(__dirname, '__fixtures__', 'scoped.js'),
-  );
-  expect(resolved).toEqual([
-    expect.stringContaining(path.join('@myorg', 'pkg')),
-  ]);
-});
+  test('resolves dependencies for scoped packages', () => {
+    const resolved = dependencyResolver.resolve(
+      path.resolve(__dirname, '__fixtures__', 'scoped.js'),
+    );
+    expect(resolved).toEqual([
+      expect.stringContaining(path.join('@myorg', 'pkg')),
+    ]);
+  });
 
-test('resolves no inverse dependencies for empty paths set', () => {
-  const paths = new Set();
-  const resolved = dependencyResolver.resolveInverse(paths, filter);
-  expect(resolved.length).toEqual(0);
-});
+  describe('includeCoreModules', () => {
+    test('does not include core modules without the option', () => {
+      const resolved = dependencyResolver.resolve(
+        path.resolve(__dirname, '__fixtures__', 'core.test.js'),
+      );
+      expect(Array.from(resolved)).toEqual([]);
+    });
 
-test('resolves no inverse dependencies for set of non-existent paths', () => {
-  const paths = new Set(['/non/existent/path', '/another/one']);
-  const resolved = dependencyResolver.resolveInverse(paths, filter);
-  expect(resolved.length).toEqual(0);
+    test('includes core modules with the option', () => {
+      const resolved = dependencyResolver.resolve(
+        path.resolve(__dirname, '__fixtures__', 'core.test.js'),
+        {includeCoreModules: true},
+      );
+      expect(Array.from(resolved)).toEqual(['fs']);
+    });
+  });
 });
 
-test('resolves inverse dependencies for existing path', () => {
-  const paths = new Set([path.resolve(__dirname, '__fixtures__/file.js')]);
-  const resolved = dependencyResolver.resolveInverse(paths, filter);
-  expect(resolved).toEqual([
-    expect.stringContaining(
-      path.join('__tests__', '__fixtures__', 'file.test.js'),
-    ),
-  ]);
+describe('resolveRecursive', () => {
+  test('resolves no dependencies for non-existent path', () => {
+    const resolved = dependencyResolver.resolveRecursive('/non/existent/path');
+    expect(Array.from(resolved)).toHaveLength(0);
+  });
+
+  test('resolves dependencies for existing path', () => {
+    const resolved = dependencyResolver.resolveRecursive(
+      path.resolve(__dirname, '__fixtures__', 'recursive', 'a.js'),
+    );
+    expect(Array.from(resolved)).toEqual([
+      expect.stringContaining('b.js'),
+      expect.stringContaining('c1.js'),
+      expect.stringContaining('c2.js'),
+    ]);
+  });
+
+  test('resolves circular dependencies', () => {
+    const resolved = dependencyResolver.resolveRecursive(
+      path.resolve(__dirname, '__fixtures__', 'circular', 'a.js'),
+    );
+    expect(Array.from(resolved)).toEqual([
+      expect.stringContaining('b.js'),
+      expect.stringContaining('c.js'),
+    ]);
+  });
+
+  describe('includeCoreModules', () => {
+    test('does not include core modules without the option', () => {
```

Comment:
looks like this whole describe block is a duplicate of https://github.com/facebook/jest/pull/7553/files?w=1#diff-1b0a8ce01682e489c73d2df8c2684095R67

Reply:
One is for `resolve`, one for `resolveRecursive` (these are 2nd level `describe`s).
I considered DRYing this up with `forEach` but I do not think it's worth for two test cases and I would find it confusing given that `resolve` and `resolveRecursive` work so differently (and would not always return the same things with `includeCoreModules` like they do in these test cases)